internal/app/machined/pkg/controllers/siderolink/manager.go

```diff
@@ -5,40 +5,55 @@
 package siderolink
 
 import (
-	"bytes"
 	"context"
 	"crypto/tls"
+	"errors"
 	"fmt"
 	"net/netip"
 	"net/url"
+	"os"
 	"regexp"
+	"time"
 
 	"github.com/cosi-project/runtime/pkg/controller"
 	"github.com/cosi-project/runtime/pkg/resource"
+	"github.com/cosi-project/runtime/pkg/safe"
 	"github.com/cosi-project/runtime/pkg/state"
 	"github.com/siderolabs/go-pointer"
-	"github.com/siderolabs/go-procfs/procfs"
 	pb "github.com/siderolabs/siderolink/api/siderolink"
 	"github.com/siderolabs/siderolink/pkg/wireguard"
 	"go.uber.org/zap"
+	"golang.zx2c4.com/wireguard/wgctrl"
 	"golang.zx2c4.com/wireguard/wgctrl/wgtypes"
 	"google.golang.org/grpc"
 	"google.golang.org/grpc/credentials"
 	"google.golang.org/grpc/credentials/insecure"
 
-	"github.com/siderolabs/talos/internal/pkg/smbios"
 	"github.com/siderolabs/talos/pkg/machinery/constants"
 	"github.com/siderolabs/talos/pkg/machinery/nethelpers"
+	"github.com/siderolabs/talos/pkg/machinery/resources/config"
+	"github.com/siderolabs/talos/pkg/machinery/resources/hardware"
 	"github.com/siderolabs/talos/pkg/machinery/resources/network"
+	"github.com/siderolabs/talos/pkg/machinery/resources/siderolink"
 )
 
 // ManagerController interacts with SideroLink API and brings up the SideroLink Wireguard interface.
 type ManagerController struct {
-	Cmdline *procfs.Cmdline
-
 	nodeKey wgtypes.Key
 }
 
+// NewManagerController creates a new ManagerController.
+func NewManagerController() (*ManagerController, error) {
+	nodeKey, err := wgtypes.GeneratePrivateKey()
```

Comment:
can we push it down to the first step of `ManagerController.Run()`? as the controller struct survives the restarts, we can see if the key is zero, and create a new one (once).

or we could simply use `sync.Once` as well.

Reply:
sure, done.